app/services/cycle_timetable.rb

```diff
@@ -243,32 +137,20 @@ def self.holidays(year = current_year)
     real_schedule_for(year).fetch(:holidays)
   end
 
-  def self.apply_1_deadline_first_reminder
-    (apply_1_deadline - 2.months).beginning_of_week + 1.week
-  end
-
-  def self.apply_1_deadline_second_reminder
-    (apply_1_deadline - 1.month).beginning_of_week + 1.week
-  end
-
-  def self.apply_2_deadline_first_reminder
-    (apply_2_deadline - 2.months).beginning_of_week + 1.week
+  def self.apply_deadline_first_reminder
+    # For 2024, date confirmed is Wednesday 17 July at 6pm
+    apply_deadline - 2.months
   end
 
-  def self.apply_2_deadline_second_reminder
-    (apply_2_deadline - 1.month).beginning_of_week + 1.week
+  def self.apply_deadline_second_reminder
+    # For 2024, date confirmed is Saturday 17 September at 6pm
+    apply_deadline - 1.month
   end
 
-  def self.between_cycles_apply_1?
-    (current_date > apply_1_deadline &&
-      current_date < apply_reopens) ||
-      current_date < apply_opens
-  end
-
-  def self.between_cycles_apply_2?
-    (current_date > apply_2_deadline &&
-      current_date < apply_reopens) ||
-      current_date < apply_opens
+  def self.between_cycles?
+    (CYCLE_DATES[next_year].present? && # We need to evaluated the next year in order to compare dates.
+      current_date.between?(apply_deadline, apply_reopens)) ||
+      current_date.before?(apply_opens)
```

Comment:
this line always confuses me. Then I realise `apply_opens` right now is 2023 apply opens. I always need to read code and notes from the past to understand when CycleTimetable.apply_opens date changes. Is it only me though that get this confusion at first hand?

Reply:
There is a lot in this class that is confusing because it is time. I will add a comment here to explain a bit better for future us.